overlays/nodejs/cloudwatch/logGroupMixins.ts

```diff
@@ -0,0 +1,148 @@
+    awslogs: {
+        // The data attribute in the Lambda record is Base64 encoded and compressed with the gzip
+        // format. Unencoded and uncompressed it will have the shape of [DecodedLogGroupEvent]
+        data: string;
+    };
+}
+
+export interface DecodedLogGroupEvent {
+    // The AWS Account ID of the originating log data.
+    owner: string;
+
+    // The log group name of the originating log data.
+    logGroup: string;
+
+    // The log stream name of the originating log data.
+    logStream: string;
+
+    // The list of subscription filter names that matched with the originating log data.
+    subscriptionFilters: string[];
+
+    // Data messages will use the "DATA_MESSAGE" type. Sometimes CloudWatch Logs may emit Lambda
+    // records with a "CONTROL_MESSAGE" type, mainly for checking if the destination is reachable.
+    messageType: string;
+
+    // The actual log data, represented as an array of log event records. The "id" property is a
+    // unique identifier for every log event.
+    logEvents: LogGroupEventRecord[];
+}
+
+export interface LogGroupEventRecord {
+    id: string;
+    timestamp: number;
+    message: string;
+}
+
+export type LogGroupEventHandler = lambda.EventHandler<LogGroupEvent, void>;
+
+export class LogGroupEventSubscription extends lambda.EventSubscription {
+    public readonly logGroup: pulumi.Output<logGroup.LogGroup>;
+    public readonly logSubscriptionFilter: logSubscriptionFilter.LogSubscriptionFilter;
+
+    constructor(
+        name: string, logGroup: logGroup.LogGroup, handler: LogGroupEventHandler,
+        args: LogGroupEventSubscriptionArgs, opts?: pulumi.ResourceOptions) {
+
+        super("aws:cloudwatch:LogGroupEventSubscription", name, { logGroup: logGroup }, opts);
+
+        args = args || {};
+        const parentOpts = { parent: this };
+        this.func = lambda.createFunctionFromEventHandler(name, handler, parentOpts);
+
+        this.permission = new lambda.Permission(name, {
+            action: "lambda:invokeFunction",
+            function: this.func,
+            principal: `logs.${config.requireRegion()}.amazonaws.com`,
+            sourceArn: logGroup.arn,
+        }, parentOpts);
+
+        this.logSubscriptionFilter = new logSubscriptionFilter.LogSubscriptionFilter(name, {
+            destinationArn: this.func.arn,
+            filterPattern: args.filterPattern || "",
+            logGroup: logGroup,
+        }, parentOpts);
+    }
+}
+
+declare module "./logGroup" {
+    interface LogGroup {
+        /**
+         * Creates a new subscription to events fired from this LogGroup to the handler provided,
+         * along with options to control the behavior of the subscription.
+         *
+         * The events will be produced in raw (gzipped + base64 encoded) form.
+         */
+        onEvent(name: string, handler: LogGroupEventHandler,
+                args?: LogGroupEventSubscriptionArgs, opts?: pulumi.ResourceOptions): LogGroupEventSubscription;
```

Comment:
bikeshedding. how does .onEvent sound to people here?   we could say .onNewLogs, but i don't think there's high value in changing hte name.